birdhouse/deployment/deploy.sh

```diff
@@ -137,10 +137,12 @@ set -x
 # stop all to force reload any changed config that are volume-mount into the containers
 ./pavics-compose.sh stop
 
-# this container is not managed by docker-compose, have to handle it manually
+# user containers are not managed by docker-compose, have to handle them manually
 # rm and not just stop to force spawning newer image
-docker stop jupyter-public
-docker rm jupyter-public
+for jupyter_cont in `docker ps --format '{{.Names}}' | grep jupyter-`; do
+    docker stop $jupyter_cont
```

Comment:
I think this is handled by the dockerspawner plugin. Everytime I restarted the jupyterhub image, the user containers got stopped and removed also. 

I'm not sure, but maybe manually stopping and removing them this way could interfere with post-stop hooks (although we don't have any right now).

Reply:
Oh !  This is a behavior change that was not there in the previous JupyterHub version we used.  I didn't notice that, thanks for bringing it up.  That's why I had to manually kill the jupyter server of the public user previously.

Not sure I like this new behavior.  With the old behavior, I can deploy a new jupyterhub (`./pavics-compose.sh up -d`) without killing jobs running in the user jupyter server.  And when the JupyterHub is back, user can resume working as if nothing happened.

`deploy.sh` runs during night time only for prod server so it was okay that it kills user jupyter server.

Reply:
You are right, `./pavics-compose.sh stop jupyterhub` kills all the user jupyter servers !  This sucks !  Wonder why this behavior change is needed !

Can not do live update of JupyterHub during the day anymore.  I used to live update JupyterHub to publish new Jupyter image, then let Travis know so he restarts his user jupyter server at his convenient time to not kill his running jobs.